pkg/api/types.go

```diff
@@ -138,6 +138,21 @@ type ObjectMeta struct {
 	// objects.  Annotation keys have the same formatting restrictions as Label keys. See the
 	// comments on Labels for details.
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// List of objects depended by this object. If ALL objects in the list have
+	// been deleted, this object will be garbage collected.
+	OwnerReferences []ObjectReference `json:"ownerReferences,omitempty"`
```

Comment:
OrdainedBy
PermittedBy
DeleteWhenAllGone
ChainDeletionOf
GCReferences
GCLockedBy
GCLockReferences

I'm not sure any of these beat OwnerReferences, maybe it'll give someone a good idea.